### PR TITLE
[vernac] Move flags commands to their own file.

### PR DESCRIPTION
**Pull request by ejgallego**
This is a small cleanup towards cleaning up vernacentries, in this
case we move setup of global flags to their own file; they were for
historical reasons, but really not related to the rest of the file.

### CODE REVIEW

File: vernac/comFlags.ml [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+    { optdepr  = false;
+      optkey   = ["Silent"];
+      optread  = (fun () -> !Flags.quiet);
+      optwrite = ((:=) Flags.quiet) }
+
+let () =
+  declare_bool_option
+    { optdepr  = false;
+      optkey   = ["Implicit";"Arguments"];
+      optread  = Impargs.is_implicit_args;
```
Comment (SkySkimmer) [inline vernac/comFlags.ml:41]: Shouldn't this go to impargs.ml instead?
AFAICT only the Global and Flags options need to be higher than the associated data

Comment (ejgallego) [reply vernac/comFlags.ml:41]: Good point; on the other hand I'm trying to move `GOptions` to `vernac` so I dunno; we should maybe discuss about the design here a bit.

Comment (ejgallego) [reply vernac/comFlags.ml:41]: Updated.

**Review by herbelin — APPROVED** [review]
It makes sense to me and I like it.

For implicit argument options, if one is moved to impargs, the 5 others of them should too. (So maybe in another PR after all.)

File: vernac/comFlags.ml (hunk 1 above)
Comment (SkySkimmer) [reply vernac/comFlags.ml]: I didn't mean just this specific option, there are a bunch more which shouldn't be in the new file.

**Comment by SkySkimmer** [comment]
>For implicit argument options, if one is moved to impargs, the 5 others of them should too. (So maybe in another PR after all.)

I prefer 1 commit, less code churn that way.

**Comment by herbelin** [comment]
> > For implicit argument options, if one is moved to impargs, the 5 others of them should too. (So maybe in another PR after all.)
> 
> I prefer 1 commit, less code churn that way.

As you like but a similar question may arise with e.g. the `Constrextern` options.

**Comment by SkySkimmer** [comment]
I wasn't being speciic to impargs so clearly the intern options should also move to intern files.

**Comment by ejgallego** [comment]
Umm, let's wait for some other work regarding state before pushing options around indeed.

**Comment by SkySkimmer** [comment]
>Umm, let's wait for some other work regarding state before pushing options around indeed.

We may as well close this and reopen when progress has happened then.

**Pull request closed without merging**